### ISSUE TITLE
5thEdition Enumeration Definition Page 133

### ISSUE THREAD

**Issue by 1w3w1** [rbwhitaker/CSharpPlayersGuideEarlyAccess#722]
On Page 133 of the 5th Edition, under the section "Defining an Enumeration", it is mentioned that
"
...
New type definitions, including enumerations, must come after our main method and the methods it owns (or in a separate file, as we will do later).
...
"

My understanding is that there is no strict rule about where the enumeration definition should live as long as it has adequate scope. For example, it can be at the class level before the main method?
Thank you!

**Comment by rbwhitaker** [comment]
This is generally true, but there are a few limitations. For example, an enumeration cannot be defined inside an enumeration. Essentially, there are two places where an enumeration can be defined: (1) at the top level, directly in a namespace (including the global namespace, outside of other explicitly defined namespaces) or (2) inside a class (or struct, or some of the other fancy types like records that are, essentially, just fancy class definitions).

You cannot, however, define an enumeration (or any other type, for that matter) inside of a method. And that's what this text is hinting at. It gets weird with top-level statements. The compiler assumes that the whole file is your main method, up until it finds proof that you cannot be in the main method anymore. Since type definitions cannot appear inside a method, the compiler knows that your main method must be over once it encounters a type definition, like an enum definition. And once the main method is over, you can't try to start it up again. So if you do this, you'll get a compiler error:
```cs
int x = 3;
enum Thing { A, B, C }
x++;
```
In most ways, this is actually more about the limitations of a top-level statement style of main method than about the enumeration definition. You can't scatter your main method code all over the place, and the enum declaration signifies to the compiler that the main method code must, inevitably, be at an end.

In short, you can put the enum almost anywhere, but in a file containing top-level statements, all type definitions, enums included, must be at the bottom to avoid slicing the main method in two.
